bmlab/controllers.py

```diff
@@ -249,3 +392,28 @@ def calculate_fwhm_f(self, time, peak_position, peak_fwhm):
                 brillouin_peak_left_slope_f
             )
         return None
+
+
+class ExtractionController(object):
+
+    def extract_calibration_spectrum(self, calib_key, frame_num=None):
```

Comment:
I would say this function should be in the extraction model, see https://github.com/BrillouinMicroscopy/bmlab/issues/22#issuecomment-833292879.

And writing the `extracted_values` to the model is not necessary, I think. As far as I know, we never read it from there.

Reply:
Then, the CalibrationController would not need the ExtractionController, but only the ExtractionModel.

Reply:
Hm... actually it is a function with logic that writes into the model.... sounds like a controller semantically...

Up to now it was in the Session class, where is definitly the wrong place.

Reply:
Yes, the session is definitely the wrong place.

> Hm... actually it is a function with logic that writes into the model.... sounds like a controller semantically...

That's what I mean, I doesn't need to write the data into the model, because we only need it once and never read it again from the model, but only through the function. So no need to be a controller.

Reply:
Nevertheless I think it would be better to put it in the controller, because the function needs access to the calibration raw data of the currently selected repetition. The model should not know about the status of the session, just the other way round, right? And putting the function in the ExtractionModel would cause a circular dependence between the session module and the extraction_model module.

Reply:
Anyways, we can still change this later. For now I will merge it into main because I don't have much time now and I don't want my changes to interfere with your work.

Reply:
Alright, thanks a lot.